pkg/scalers/prometheus_scaler.go

```diff
@@ -89,17 +90,32 @@ func NewPrometheusScaler(config *ScalerConfig) (Scaler, error) {
 
 	httpClient := kedautil.CreateHTTPClient(config.GlobalHTTPTimeout, meta.unsafeSsl)
 
-	if meta.prometheusAuth != nil && (meta.prometheusAuth.CA != "" || meta.prometheusAuth.EnableTLS) {
-		// create http.RoundTripper with auth settings from ScalerConfig
-		transport, err := authentication.CreateHTTPRoundTripper(
-			authentication.NetHTTP,
-			meta.prometheusAuth,
-		)
+	if meta.prometheusAuth != nil {
+		if meta.prometheusAuth.CA != "" || meta.prometheusAuth.EnableTLS {
+			// create http.RoundTripper with auth settings from ScalerConfig
+			transport, err := authentication.CreateHTTPRoundTripper(
+				authentication.NetHTTP,
+				meta.prometheusAuth,
+			)
+			if err != nil {
+				logger.V(1).Error(err, "init Prometheus client http transport")
+				return nil, err
+			}
+			httpClient.Transport = transport
+		}
+	} else {
```

Comment:
Doesn't this mean that if no auth is configured, it assumes Azure Managed Prometheus is required?

We might need an `authenticationType` field imo

Reply:
@tomkerkhove It doesn't assume that it is managed prometheus. In case of no auth, it checks if it is so, and **tries** to get a round tripper. If it gets, it uses it otherwise proceed with no error. Does that seem ok?

Reply:
My bad!